src/_nebari/stages/infrastructure/__init__.py

```diff
@@ -174,6 +174,7 @@ class AWSInputVars(schema.Base):
     eks_endpoint_access: Optional[
         Literal["private", "public", "public_and_private"]
     ] = "public"
+    eks_kms_arn: Optional[str] = None
```

Comment:
I think this field should be immutable to prevent clusters from being deleted.

See [src/_nebari/stages/kubernetes_services/__init__.py:66 ](https://github.com/nebari-dev/nebari/blob/88dfe2447c56689c3a9828e6abef4e1c02903faf/src/_nebari/stages/kubernetes_services/__init__.py#L66) for an example of an immutable field.

Reply:
> So @joneszc am I reading that correctly that enabling this option will destroy and replace your cluster? We should probably go ahead and make this field immutable then. We definitely don't want anyone accidentally destroying their deploy. The docs should reflect that this should only be used on fresh deploys too.

Hello @dcmcand 
Enabling the `eks_kms_arn` option, including doing so as a re-deploy on an existing cluster, is not a problem.  The issue arises when one attempts to disable the cluster encryption on an existing cluster, which causes a failed attempt to destroy the cluster.  I've added a warning for this behavior in the [docs update](https://github.com/nebari-dev/nebari-docs/pull/537/commits/2bbe698ea4826607c9d450401d4c81e5c73c3fa6).  Making the field immutable will block users from being able to set cluster encryption on an existing cluster, which is functional. Is there a mechanism for allowing the field to be set, if not set already, but immutable if altered from a set state?  The risk of destroying the cluster is ineluctably countered by the error:
`Error: deleting EKS Cluster (nebari-test-dev): operation error EKS: DeleteCluster, https response error StatusCode: 409, RequestID: de8f18ba-0abe-42ae-961f-86d8865fbcf3, ResourceInUseException: Cluster has nodegroups attached`

Reply:
@joneszc can you test the above scenario and update with the latest main?

Reply:
@dcmcand 
Yes, that is the correct logic flow.  I'm testing now.  I need to test how the components behave as well after enabling encryption and then actually encrypting the secrets, as the docs mention [this important step](https://docs.aws.amazon.com/eks/latest/userguide/enable-kms.html#:~:text=After%20you%20enabled), which might have additional repercussions on Nebari components besides the Cluster functionality.

Reply:
@dcmcand 
Following a failed attempt to disable encryption, by setting a previously configured `amazon_web_services.eks_kms_arn` value to `null` or removing the field altogether, a follow-up re-deploy to reenable the original encryption setting does succeed without any issues.  Also, an attempt to switch KMS keys by replacing a previously set KMS key arn with a new arn will succeed in a re-deploy, but the KMS key is not actually changed on the EKS Cluster.   Essentially, no change is made and the cluster functions as before.  I'll make a note of this in the docs PR.

Reply:
That's good to hear, @joneszc. If I understood correctly, if by any chance the user removes the encryption key from the config and redeploys, the user will first face an error, but what I assume a few minutes, the user may be able to retain the redeployment process and succeeding in reaching the same state before the key by re-attempting the deployment? 

If so, please add this as an [admonition](https://docusaurus.io/docs/markdown-features/admonitions) to the docs under a `warning`.